src/core/reopt.jl

```diff
@@ -652,14 +741,14 @@ function add_variables!(m::JuMP.AbstractModel, p::REoptInputs)
 		# TODO: currently defining more decision variables than necessary b/c using rectangular arrays, could use dicts of decision variables instead
         @variables m begin # if there is more than one specified outage, there can be more othan one outage start time
 			dvUnservedLoad[S, tZeros, outage_time_steps] >= 0 # unserved load not met by system
-			dvMGProductionToStorage[p.techs.elec, S, tZeros, outage_time_steps] >= 0 # Electricity going to the storage system during each time_step
-			dvMGDischargeFromStorage[S, tZeros, outage_time_steps] >= 0 # Electricity coming from the storage system during each time_step
-			dvMGRatedProduction[p.techs.elec, S, tZeros, outage_time_steps]  # MG Rated Production at every time_step.  Multiply by production_factor to get actual energy
-			dvMGStoredEnergy[S, tZeros, 0:max_outage_duration] >= 0 # State of charge of the MG storage system
+			dvMGProductionToStorage[union(p.s.storage.types.elec, p.s.storage.types.hydrogen), union(p.techs.elec, p.techs.electrolyzer, p.techs.compressor), S, tZeros, outage_time_steps] >= 0 # Electricity going to the storage system during each time_step
```

Comment:
This allows non-physical flows from elec techs to H2 storage and from electrolyzer and compressor to electric storage. I can't find anywhere in the code where these flows are fixed to 0. Yes, they aren't included in the SOC equations, but they should still be fixed to 0. For one, I'm not confident that there aren't other places it could cause inaccuracies. Second, I think it could improve solve time to reduce the solution space. All of this also goes for the non-MG version of the variable.

Reply:
compressor or electrolyzer to h2 storage should also be fixed to 0 depending on require compression